pkg/syncer/syncer.go

```diff
@@ -282,11 +282,25 @@ func StartSyncer(ctx context.Context, cfg *SyncerConfig, numSyncerThreads int, i
 			}
 
 			// Start and sync informer factories
+			waitForCacheSyncContext, cancelWaitForCacheSync := context.WithCancel(ctx)
+			defer cancelWaitForCacheSync()
+			var waitForCacheSyncErrorCount int
 			var cacheSyncsForAlwaysRequiredGVRs []cache.InformerSynced
+			waitForCacheSyncErrors := map[schema.GroupVersionResource]map[string]int{}
 			for _, alwaysRequiredGVR := range alwaysRequiredGVRs {
 				if informer, err := ddsifForUpstreamSyncer.ForResource(alwaysRequiredGVR); err != nil {
 					return nil, err
 				} else {
+					errors := make(map[string]int)
+					_ = informer.Informer().SetWatchErrorHandler(func(r *cache.Reflector, err error) {
+						errStr := err.Error()
+						errors[errStr]++
```

Comment:
I'm not 100% certain but I am pretty sure that the watch error handling can happen in a different goroutine, so we would need some mutexes around this.

Reply:
:+1:

Reply:
This code block was removed.